src/modules/job-manager/restart.c

```diff
@@ -220,7 +222,8 @@ int restart_save_state_to_txn (struct job_manager *ctx, flux_kvs_txn_t *txn)
     if (flux_kvs_txn_pack (txn,
                            0,
                            checkpoint_key,
-                           "{s:I s:O}",
+                           "{s:i s:I s:O}",
+                           "version", CHECKPOINT_VERSION,
```

Comment:
This hunk can be deleted and  setting the version key can be introduced with version 1.    The version 0 in the wild does not set the version key so we shouldn't start doing that here.